tests/baselines/reference/callChain.3.types

```diff
@@ -45,9 +45,9 @@ const n4: number | undefined = a?.m?.({x: absorb()}); // likewise
 >a?.m : (<T>(obj: { x: T; }) => T) | undefined
 >a : { m?<T>(obj: { x: T; }): T; } | undefined
 >m : (<T>(obj: { x: T; }) => T) | undefined
->{x: absorb()} : { x: number | undefined; }
->x : number | undefined
->absorb() : number | undefined

```

Comment:
Interesting change to inference - I assume this is because previously this `undefined` was the optional chain type, and so managed to skip out on the "cleave matching types" logic in inference?

Reply:
That's correct. The resulting type now aligns with other examples in the file.